src/formatDate.js

```diff
@@ -50,7 +50,48 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newDate = date;
+  const dateArray = newDate.split(fromFormat[3]);
+  const newDateArray = [];
+  let yearIndex = 0;
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    yearIndex = fromFormat.indexOf('YY');
+
+    const year = dateArray[yearIndex];
+
+    if (year < 30) {
+      dateArray[yearIndex] = `20${year}`;
+    } else {
+      dateArray[yearIndex] = `19${year}`;
+    }
+  } else if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    yearIndex = fromFormat.indexOf('YYYY');
+
+    dateArray[yearIndex] = dateArray[yearIndex].slice(-2);
+  }
+
+  for (let i = 0; i < 3; i++) {
+    let desired = toFormat[i];
+
+    if (desired === 'YY' && !fromFormat.includes(desired)) {
+      desired = 'YYYY';
+    } else if (desired === 'YYYY' && !fromFormat.includes(desired)) {
+      desired = 'YY';
+    }
+
+    yearIndex = fromFormat.indexOf(desired);
+
+    newDateArray[i] = dateArray[yearIndex];
+  }
+
+  if (fromFormat[3] !== toFormat[3]) {
```

Comment:
do we really need check here?
i guess, if they equal or different u can just always use new separator